stage0/src/main.rs

```diff
@@ -190,18 +198,15 @@ pub extern "C" fn rust64_start(encrypted: u64) -> ! {
      * See https://www.kernel.org/doc/html/latest/x86/boot.html#id1 for the particular requirements.
      */
 
-    // Safety, for all these data structure dereferences: The addresses are to well-known locations,
-    // same as crosvm is using, in the lowest 1MB of memory which we know is valid.
-    let gdt = unsafe { &mut *(BOOT_GDT_ADDR.as_u64() as *mut GlobalDescriptorTable) };
-
     if snp {
         pvalidate(
-            x86_64::addr::align_down(gdt as *const _ as u64, 0x1000) as usize,
+            x86_64::addr::align_down(unsafe { BOOT_GDT.as_ptr() } as u64, 0x1000) as usize,
```

Comment:
Please add a comment somewhere in the code (e.g. in module-level doc commentss) to explain why these many unsafe blocks are actually ok, similar to what is in the PR description.

Reply:
Done.